src/kernels/MIOpenPadReflection.cpp

```diff
@@ -0,0 +1,126 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <cstddef>
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
+#include <hip/hip_fp16.h>
+#include <hip/hip_runtime.h>
+#endif
+
+#include "float_types.h"
+
+template <typename TI, typename TO>
+__device__ void padReflection2dFwdContiguous(const TI* __restrict__ input,
+                                             TO* __restrict__ output,
+                                             uint64_t output_size,
+                                             long padding_l,
+                                             long padding_t,
+                                             const size_t in_H,
+                                             const size_t in_W,
+                                             const size_t output_size_1,
+                                             const size_t output_size_2,
+                                             const size_t output_size_3,
+                                             const size_t input_stride_0,
+                                             const size_t input_stride_1,
+                                             const size_t input_stride_2,
+                                             const size_t input_stride_3)
+{
+    const size_t gid = threadIdx.x + blockIdx.x * blockDim.x;
+    if(gid >= output_size)
+        return;
+
+    long n, c, h, w;
+    ulong nch = (gid) / output_size_3;
+    w         = (gid) % output_size_3;
+    ulong nc  = nch / output_size_2;
+    h         = nch % output_size_2;
+    n         = nc / output_size_1;
+    c         = nc % output_size_1;
+
+    long in_start_x  = max(0L, -padding_l);
+    long in_start_y  = max(0L, -padding_t);
+    long out_start_x = max(0L, padding_l);
+    long out_start_y = max(0L, padding_t);
+
+    if(w < padding_l)
+    {
+        w = padding_l * 2 - w;
+    }
+    else if(padding_l <= w && w < in_W + padding_l)
+    {
```

Comment:
You can add a condition to line 74 and remove this empty else if.

Reply:
resolved in https://github.com/moreh-dev/MIOpen/pull/7/commits/c916b1d4d45978492def1503f2539eb7c1caec4c